### PR TITLE
Solves 2909  "Requesting category check" notification even though I did not mark as such 

### LINKED ISSUE #2909: "Requesting category check" notification even though I did not mark as such
Categorization of the picture was correct, so I pushed the button accordingly, but despite that a "Requesting category check" notification is shown. I have not checked whether the request is actually uploaded or not.
Screenshot_20190422-125319

### PR DESCRIPTION
**Pull request by thetimeheist76**
Category check was requested when sending thanks to the contributor even though the category was not reported to be wrong by the reviewer.

Fixes #2909 "Requesting category check" notification even though I did not mark as such 

Changes made:

- There was a call for building the notification of "check category" while sending thanks which is not required. When a category check is requested the "reportwrongcategory" method is directly called, hence a call to publish progress for check category in the sendthanks method is not needed. I've changed the 'i' value that is sent with publishprogress to '1' as '2' corresponds to check category.

- I've added a condition in the publishprogress method which leads to checking whether a category check was requested and only then publish a "category check" notification. This prevents publishing the notification when only thanks is needed to be sent.

Tests performed 👍 
When category check is not requested by the reviewer and he sends thanks to the contributor category check is not requested.
Android version 7
API level supported: 19-21

Screenshots showing what changed:
Before:

After:
sending thanks ss

Need help? See https://support.google.com/android/answer/9075928

---

_Note: Please ensure that you have read CONTRIBUTING.md if this is your first pull request._

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java [hunk 1]
```diff
@@ -3,6 +3,8 @@
 import android.annotation.SuppressLint;
 import android.content.Context;
 import android.content.Intent;
+import android.graphics.PorterDuff;
```
Comment (ashishkumar468) [inline app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java]: This file has no relevant changes, please roll back the formatting changes

File: app/src/main/java/fr/free/nrw/commons/review/ReviewController.java [hunk 2]
```diff
@@ -156,13 +156,14 @@ private void publishProgress(@NonNull Context context, int i) {
             message = context.getString(messages[i]);
         }
 
-        notificationBuilder.setContentTitle(context.getString(R.string.check_category_notification_title, media.getDisplayTitle()))
+        if(i==2)// will build the notification for category check only if i==2
```
Comment (ashishkumar468) [inline app/src/main/java/fr/free/nrw/commons/review/ReviewController.java]: Can you add a more descriptive comment :)

**Comment by thetimeheist76** [comment]
I had to uninstall android studio from my machine due to technical problems. I am so sorry I wont be able to work on this.

**Comment by neslihanturan** [comment]
Closing this one since the owner said they are nit interested anymore and the pr has already some problems.

**Pull request closed without merging**